### DISCUSSION TITLE
Cross validation used in FLAML

### DISCUSSION THREAD

**Discussion by hanhanwu** [microsoft/FLAML#154] [Q&A]
Hello,

I just read FLAML paper and have 4 questions :)

1. If the resampling strategy propser will decide to use cross validation or holdout, does it mean even if in `settings` I chose to use holtout, FLAML might still use CV when it thinks CV will be better than holdout?
2. When we are using `automl`, does it use Economical Hyperparameter Optimization? If so, by default it's CFO or BlendSearch?
3. In hyperparam and sample size propser, it mentioned data will be randomly shuffled. Does this mean we cannot use FLAML for time series data when we have to keep the order of the data records? 
4. For the cross validation settings, are we allowed to use time series split similar to this in sklearn https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html ? I'm mainly trying to figure out whether we can use FLAML for time series data, when the data order need to be kept

Thank you very much!

**Comment by sonichi** [comment]
Hi @hanhanwu 

1. If you choose `holdout` explicitly, your choice will be respected. FLAML only decides cv/holdout when `eval_method='auto'`.
2. Yes it does. It uses CFO by default as the default search space is relatively simple.
3. Support for time series data is work-in-progress and is close to be released. Can we go over your use case and understand your requirements? Feel free to chat on [gitter](https://gitter.im/FLAMLer/community).
4. Yes, it will be part of the upcoming support for time series data. Looking forward to the discussion.

**Reply by hanhanwu** [reply]
Sounds great!

And what will be the right way to choose "blendsearch" to automl settings? I tried to set "bs" for `hpo_method` but got this warning:
![bs_warning](https://user-images.githubusercontent.com/4024769/128641428-77b65b4c-ad6c-412d-81e5-0698f51e395a.PNG)
Now I'm not sure whether I'm using blendsearch correctly...

For time series, currently we are using sklearn time series split to make sure our data is in time order in each fold. Will look forward to see the updates from FLAML. More questions, I will ask in gitter in the future. Thank you very much!
